flycheck.el

```diff
@@ -7735,22 +7749,20 @@ contains a cabal file."
     (`haskell-stack-ghc
      (or
       (when (buffer-file-name)
-        (locate-dominating-file (buffer-file-name) "stack.yaml"))
+        (flycheck--locate-dominating-file-matching
+         (file-name-directory
```

Comment:
`(file-name-directory` is superfluous for `locate-dominating-file`, why not do the same for our new function?